docs/specifications/4-standard-compliance-suite.md

```diff
@@ -0,0 +1,166 @@
+# Standard Compliance Suite
+
+## Format
+In order to assert whether a new runtime implementation is correct, a formal
+compliance test suite is defined, that all new language runtimes must fully
+implement before they can be deemed eligible for General Availability.
+
+This document describes these tests, as well as a general approach for ensuring
+conformance requirements are met in a systematic manner.
+
+### Categories
+Test cases in the standard compliance suite are grouped by categories, which
+help implementors direct their effort in the early stages of the implementation
+of new language bindings. Each category is declared in an `H3` title (a line
+that starts with `### `) within the [`## Test Suite`] title. A description of
+the category immediately follows the opening title. The category ends with the
+end of the document, or whenever another `H2` title is reached.
+
+[`## Test Suite`]: #test-suite
+
+### Test Case
+Within a category title, test cases are delimited by `H4` (`#### `) titles,
+which correspond to the test case name. The test case name should be kept
+concise (dieally within 75 characters) and try to be as descriptive as possible.
+
+Immediately after the `H4` title is an english language description of the test
+case that explains the property the test is designed to validate in as much
+detail as possible. As much as possible, test case descriptions should be
+self-sufficient.
+
+After the description, a table summarizes information about the test case that
+can be useful when investigating the test:
+- **Introduced** specifies the very first *released* version of `jsii` that
+  included the test. This implies previous versions of `jsii` may not have
+  conformed to the test.
+- **Related Issues** lists any and all issues related to the test case. This
+  includes the issue that was created to track the introduction of the test, and
+  other issues such as bugs that prompted the introduction of a new compliance
+  test.
+- **Revised** lists all versions of `jsii` after the initial introduction of the
+  test case, for which the release included a revision of the test case. Trivial
+  revisions (such as correcting typos, or re-phrasing the description without
+  changing the tests case's semantic value) should not be included in the list.
+
+After the attributes table, a **TypeScript** block of code describes the
+canonical form of the test. It includes any type declaration that is used by the
+test (so the code example is self-contained). Assertions performed by the test
+should be written in the form of [`jest`] expectations.
+
+[`jest`]: https://jestjs.io/docs/en/getting-started
+
+Finally, another code block details the sequence of messages that should be
```

Comment:
Is this checked or does it mostly serve as a hint to implementors/debuggers?

Reply:
Is it going to including kernel initialization, or only the communication after the module has been loaded?

Reply:
The sequence of messages is normative, it is necessary to ensure consistent behavior across languages, at least from a side-effects perspective.

The initialization *might* be truncated off, for sure (it seems very inconvenient indeed).